bin/e2hs-writer/src/reader.rs

```diff
@@ -197,33 +176,18 @@ impl EpochReader {
         let transactions = decode_transactions(&payload.transactions)?;
         let withdrawals: Vec<Withdrawal> =
             payload.withdrawals.iter().map(Withdrawal::from).collect();
-        let header =
-            pre_deneb_execution_payload_to_header(payload.clone(), &transactions, &withdrawals)?;
-
-        let slot = block.slot;
-
-        // create beacon block proof
-        let historical_batch_proof = era
-            .historical_batch
-            .build_block_root_proof(slot % EPOCH_SIZE);
-        let beacon_block_proof: FixedVector<B256, typenum::U13> = historical_batch_proof.into();
-
-        // create execution block proof
-        let mut execution_block_hash_proof = block.body.build_execution_block_hash_proof();
-        let body_root_proof = block.build_body_root_proof();
-        execution_block_hash_proof.extend(body_root_proof);
-        let execution_block_proof: VariableList<B256, typenum::U12> =
-            execution_block_hash_proof.into();
-        let proof = BlockProofHistoricalSummaries {
-            beacon_block_proof,
-            beacon_block_root: block.tree_hash_root(),
-            slot,
-            execution_block_proof,
-        };
 
         let header_with_proof = HeaderWithProof {
-            header,
-            proof: BlockHeaderProof::HistoricalSummaries(proof),
+            header: pre_deneb_execution_payload_to_header(
+                payload.clone(),
```

Comment:
```suggestion
            header: pre_deneb_execution_payload_to_header(
                payload,
```

Do we need this clone?